src/sentry/models/notificationsetting.py

```diff
@@ -91,7 +93,9 @@ class NotificationSetting(Model):
     )
     # user_id, organization_id, project_id
     scope_identifier = BoundedBigIntegerField(null=False)
-
+    target = FlexibleForeignKey(
+        "sentry.Actor", db_index=True, unique=False, null=False, on_delete=models.PROTECT
```

Comment:
`on_delete` defines what we'll do here if the `Actor` is deleted. I think in this case we'd want to delete any related `NotificationSetting` rows, rather than prevent the `Actor` from being deleted, does that sound right?

Reply:
Ah oops I was thinking about it the other way around - I didn't want the user to be deleted if a setting was.